File: python/examples/get_compliance_events.py
import pprint
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from request import BaseRequest
from dotenv import dotenv_values

config = dotenv_values()

# Example of GET /compliance-events

class ComplianceEventsRequest(BaseRequest):
    def __init__(self):
        super().__init__()
    
    def get_compliance_events(self, start_date, end_date, limit=40):
        path = "/compliance-events"
        params = { 'start_date': start_date, 'end_date': end_date, 'limit': limit }
        return self.make_request("GET", path, params=params)

# run as standalone script by passing any command line argument
if len(sys.argv) > 1:
    start_date = config['START_DATE']
    end_date = config['END_DATE']

    compliance_request = ComplianceEventsRequest()
    compliance_events = compliance_request.get_compliance_events(start_date, end_date)

    pprint.pprint(compliance_events)

File: javascript/examples/get_compliance_events.js
const baseRequest = require('../base_request.js').baseRequest

const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

//  Example of GET /compliance-events

//  Either a company or company_id may be provided, but not both
//  Either an array of jurisdictions or jurisdiction_ids may be provided, but not both
//  Start_date and end_date are required

const COMPANY_ID	= process.env.COMPANY_ID;
const START_DATE 	= process.env.START_DATE;
const END_DATE    = process.env.END_DATE;

params = {
  company_id: COMPANY_ID,
  start_date: START_DATE,
  end_date: END_DATE
}

token = baseRequest.request.token({ path: '/compliance-events' });

baseRequest.request.get({ path: '/compliance-events', token: token, queryParams: params });
